src/zenohUTransport.cpp

```diff
@@ -0,0 +1,600 @@
+                //TODO - print the URI that failed 
+                spdlog::error("z_undeclare_publisher failed");
+                status.set_code(UCode::INVALID_ARGUMENT);
+                return status;
+            }
+        }
+
+        pubHandleMap_.clear();
+
+        for (auto listenerInfo : listenerMap_) {
+            for (auto sub : listenerInfo.second->subVector_) {
+                if (0 != z_undeclare_subscriber(z_move(sub))) {
+                    spdlog::error("z_undeclare_publisher failed");
+                    status.set_code(UCode::INVALID_ARGUMENT);
+                    return status;
+                } else {
+                    spdlog::debug("z_undeclare_subscriber done");
+                }
+             }
+             
+            listenerInfo.second->listenerVector_.clear();
+            listenerInfo.second->subVector_.clear();
+        }
+
+    
+        listenerMap_.clear();
+
+        if (UCode::OK != ZenohSessionManager::instance().term()) {
+            spdlog::error("zenohSessionManager::instance().term() failed");
+            status.set_code(UCode::INTERNAL);
+            return status;
+        }
+
+        spdlog::info("ZenohUTransport term done");
+    }
+
+    status.set_code(UCode::OK);
+
+    return status;
+}
+
+UStatus ZenohUTransport::send(const UUri &uri,
+                              const UPayload &payload,
+                              const UAttributes &attributes) noexcept {
+    UStatus status;
+
+    if (0 == refCount_) {
+        spdlog::error("ZenohUTransport is not initialized");
+        status.set_code(UCode::UNAVAILABLE);
+        return status;
+    }
+
+    if (true == termPending_) {
+        spdlog::error("ZenohUTransport is marked for termination");
+        status.set_code(UCode::UNAVAILABLE);
+        return status;
+    }
+    
+    if (false == uri.getUResource().isRPCMethod()) {
+        status.set_code(sendPublish(uri, payload, attributes));
+    } else {
+        status.set_code(sendQueryable(uri, payload, attributes));
+    }
+
+    return status;
+}
+UCode ZenohUTransport::sendPublish(const UUri &uri, 
+                                   const UPayload &payload,
+                                   const UAttributes &attributes) noexcept {
+    UCode status = UCode::UNAVAILABLE;
+
+    do {
+        if (UMessageType::PUBLISH != attributes.type()) {
+            spdlog::error("Wrong message type = {}", static_cast<int>(attributes.type()));
+            return UCode::INVALID_ARGUMENT;
+        }
+
+        // Serializing UAttributes
+        size_t attrSize = attributes.ByteSizeLong();
+        std::vector<uint8_t> serializedAttributes(attrSize);
+        if (!attributes.SerializeToArray(serializedAttributes.data(), attrSize)) {
+            spdlog::error("SerializeToArray failure");
+            return UCode::INTERNAL;
+        }
+
+        std::vector<uint8_t> payloadData(payload.data(), payload.data() + payload.size());
+
+        std::vector<uint8_t> finalPayload;
```

Comment:
Why did you do that?
this reverts all the work that you did in your previous PR.
you should send the payload and the header separately , and not in on buffer